src/app/public/src/modules/shared/search-results-provider.ts

```diff
@@ -0,0 +1,129 @@
+import { Injectable } from '@angular/core';
+import { SkyAuthHttp } from '@blackbaud/skyux-builder/runtime';
+import { StacheConfigService } from './config.service';
+import { SearchResult } from './search-result';
+import 'rxjs/add/operator/toPromise';
+
+@Injectable()
+export class StacheSearchResultsProvider {
+
+  private static readonly STACHE_QUERY_URL: string = 'https://stache-search-query.sky.blackbaud.com';
+  private static readonly STACHE_QUERY_ENDPOINTS: any = {
+    internal: 'query',
+    public: 'query-public'
+  };
+  private isInternal: boolean = true;
+  private siteNames: string[] = [];
+  private endpoints: any;
+  private url: string;
+
+  constructor(
+    private http: SkyAuthHttp,
+    private config: StacheConfigService
+  ) {
+    this.initIsInternal(this.config);
+    this.initEndpoints(this.config);
+    this.initSiteNames(this.config);
+    this.initUrl(this.config);
+  }
+
+  public getSearchResults(searchArgs: any): Promise<any> {
```

Comment:
Does this return a `Promise<StacheSearchResult[]>`?

Reply:
No. The stache search result is what comes from my service. The return from that is the interface that skyux established for the search provider.

Reply:
Understood!